### DISCUSSION TITLE
How do you prevent the default behavior of TYPO3 when saving sys_category?

### DISCUSSION THREAD

**Discussion by ghost** [georgringer/news#2376] [Q&A]
Following Problem:
TYPO3 deletes categories assigned to a data record or page if the user edit/saves the data record/page, but has no rights for these categories.
An Example:
The backend user has rights for a page and only categorie1. If the page has categorie1 and categorie2 and the user edit/saves the page, categorie2 is lost.

Can you please explain to me exactly how your extension bypasses this behavior. With news the whole thing works without any problems. What are you doing differently?

Thank you very much.

**Comment by georgringer** [comment]
TBH I am not sure if I do it differently. does the user have access to the category field in news and page?

**Comment by ghost** [comment]
Yes, but the user can only see and use selected categories. This is set in his user group. However, if the news has a category that he does not see, this category remains selected even after saving. I think this is great, because with all other extensions and also in TYPO3 itself this does not work. Testet with TYPO3 10,11,12, own extensions and calendarize.

**Comment by ghost** [comment]
I finally found your secret. ;-)
It's in your DataHandlerHook.php and the function processDatamap_preProcessFieldArray contains the magic. 
You combine the forbidden, invisible categories with those that the user wants to set and then save everything in the database. This is super easy and I still wonder why TYPO3 can't do this itself. Thank you and I will include this for all my extensions.
